### DISCUSSION TITLE
sensore temperatura  integrato in Alexa via node red in HA

### DISCUSSION THREAD

**Discussion by supermemo007** [indomus/forum#675] [HUB: Home Assistant]
Ciao a tutti forse ne avete gia parlato nel forum ma non riesco a trovare discussioni precendenti..... mi scuso se è un duplicato. vi spiego quello che vorrei fare.
Ho collegato i sensori di temperatura Aqara in HA usando ZHA. Tuto funziona alla grande. Vorrei ora far leggere ad Alexa i valori di temperatura di questi sensori. Per prima cosa ho creato i device su nodered utilizzando Node-RED Alexa home skill Bridge via web (creati come Thermostat) Node red in HA mi trova i device partendo dal nodo Alexa-home ma non capisco poi come procedere con il flow , sicuramente da utilizzare "current state" ma poi? è l'approccio giusto? Naturalmente vorrei che Alexa mi leggesse il valore della temperatura corrente del sensore richiamato.
Ho creato altri device in questo modo, switch e light che funzionano abbinati alle routine di Alexa riesco ad azionarli.

Grazie per il vostro supporto e per i vostri consigli !
![image](https://user-images.githubusercontent.com/66065377/131869329-62d959d0-97fa-46b9-82bc-d7a7864a1dd0.png)
![image](https://user-images.githubusercontent.com/66065377/131869366-4dcabe0a-a696-481a-8ba1-d802d76744f0.png)

**Comment by xyzmauro** [comment]
Ciao, provo a farti una "guida" step-by-step perchè effettivamente si trova poco in giro...
dopo aver creato il fake-switch sul sito ( basta solo ON ) lo fai cercare tra i dispositivi dall'app Alexa
![Alexa_1](https://user-images.githubusercontent.com/56839356/131895538-d696e338-af06-4880-96e7-cdbaa09bfdfb.png)
Crei la routine sull'app, ad esempio così
![Alexa_2](https://user-images.githubusercontent.com/56839356/131896157-bb35e6ed-780b-43d1-a449-e1f1bca58571.jpg)
A questo punto puoi costruirti il flow su Node-Red, chiedendo "Alexa, test temperatura" ti risponde con la temperatura che legge nel nodo current_state...
[flows.txt](https://github.com/indomus/forum/files/7101031/flows.txt)
devi rinominare flow.txt in flow.json prima di importarlo in Nodered perchè non mi fa allegare direttamente il file...
Spero di esser stato di aiuto :-)

**Reply by xyzmauro** [reply]
Chiaro, modifica coi tuoi sensori/nomi :-p

**Reply by supermemo007** [reply]
Ciao Grazie, ho importato il flow cambiando i nomi ed il mio sensore ma non ricevo la lettura della temperatura, la routine parte e vedo il trigger in nodered ma alexa non mi riporta la temperatura. Ho provato a creare anche il sensore come termostato ed abilitare la lettura della temp ma nulla. Qualche idea?

**Reply by xyzmauro** [reply]
Metti un nodo di debug a valle del current state e vedi il payload che esce da li

**Reply by supermemo007** [reply]
3/9/2021, 12:39:09node: 84cac997236221d9
sensor.lumi_lumi_weather_ea59da01_temperature : msg.payload : number
27.3

ho provato anche con lo state type: string

3/9/2021, 12:40:57node: 84cac997236221d9
sensor.lumi_lumi_weather_ea59da01_temperature : msg.payload : string[4]
"27.3"

funziona, ho usato un nodo notify per far leggere l'output del sensore e far partire il messaggio su Alexa, cosi funziona !
![image](https://user-images.githubusercontent.com/66065377/131993865-69cd1693-517d-4466-83d9-c0b470350c8d.png)

**Reply by xyzmauro** [reply]
Ciao, scusami ma evidentemente ieri ero ubriaco e ho sbagliato ad esportare il flusso completo, eccolo ;-)
[flows.txt](https://github.com/indomus/forum/files/7107781/flows.txt)

**Reply by supermemo007** [reply]
👍 Grazie !
